traffic_ops/app/lib/API/Configs/ApacheTrafficServer.pm

```diff
@@ -0,0 +1,1959 @@
+			}
+		}
+	}
+
+	foreach my $file ( keys %$data_obj->{'config_files'} ) {
+		if ( !defined( $data_obj->{'config_files'}->{$file}->{'scope'} ) ) {
+			$data_obj->{'config_files'}->{$file}->{'scope'} = $self->get_scope($file);
+		}
+	}
+
+	my $file_contents = encode_json($data_obj);
+
+	return $self->render( text => $file_contents, format => 'txt' );
+}
+
+#entry point for server scope api route.
+sub get_server_config {
+	my $self     = shift;
+	my $filename = $self->param("filename");
+	my $id       = $self->param('id');
+	my $scope    = $self->get_scope($filename);
+
+	##check user access
+	if ( !&is_oper($self) ) {
+		return $self->forbidden();
+	}
+
+	##check the scope - is this the correct route?
+	if ( $scope ne 'server' ) {
+		return $self->alert( "Error - incorrect file scope for route used.  Please use the " . $scope . " route." );
+	}
+
+	##verify that a valid server ID has been used
+	my $server_obj = $self->server_data($id);
+	if ( !defined($server_obj) ) {
+		return $self->not_found();
+	}
+
+	#generate the config file using the appropriate function
+	my $file_contents;
+	if ( $filename eq "12M_facts" ) { $file_contents = $self->facts( $server_obj, $filename, $scope ); }
+	elsif ( $filename =~ /to_ext_.*\.config/ ) { $file_contents = $self->to_ext_dot_config( $server_obj, $filename, $scope ); }
+	elsif ( $filename eq "ip_allow.config" ) { $file_contents = $self->ip_allow_dot_config( $server_obj, $filename, $scope ); }
+	elsif ( $filename eq "parent.config" ) { $file_contents = $self->parent_dot_config( $server_obj, $filename, $scope ); }
+	elsif ( $filename eq "records.config" ) { $file_contents = $self->generic_server_config( $server_obj, $filename, $scope ); }
+	elsif ( $filename eq "remap.config" ) { $file_contents = $self->remap_dot_config( $server_obj, $filename, $scope ); }
+	elsif ( $filename eq "hosting.config" ) { $file_contents = $self->hosting_dot_config( $server_obj, $filename, $scope ); }
+	elsif ( $filename eq "cache.config" ) { $file_contents = $self->cache_dot_config( $server_obj, $filename, $scope ); }
+	elsif ( $filename eq "packages" ) {
+		$file_contents = $self->get_package_versions( $server_obj, $filename, $scope );
+		$file_contents = encode_json($file_contents);
+	}
+	elsif ( $filename eq "chkconfig" ) {
+		$file_contents = $self->get_chkconfig( $server_obj, $filename, $scope );
+		$file_contents = encode_json($file_contents);
+	}
+	else {
+		my $file_param = $self->db->resultset('Parameter')->search( [ config_file => $filename ] )->single;
+		if ( !defined($file_param) ) {
+			return $self->not_found();
+		}
+		$file_contents = $self->take_and_bake_server( $server_obj, $filename, $scope );
+	}
+
+	#if we get an empty file, just send back an error.
+	if ( !defined($file_contents) ) {
+		return $self->not_found();
+	}
+
+	#return the file contents for fetch and db actions.
+	return $self->render( text => $file_contents, format => 'txt' );
```

Comment:
shouldn't all the returns look more like this for example? 

		$self->res->headers->content_type("application/download");
		$self->res->headers->content_disposition("attachment; filename=\"$filename\"");
		$self->render( text => $text, format => 'txt' );